terraform/product/large_deployment/variables.tf

```diff
@@ -0,0 +1,89 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+variable "cluster_name" {
+  type        = string
+  description = "The cluster name of the fleet."
+  default     = "opensearch"
+}
+
+variable "main" {
+  description = "Main orchestrator app definition"
+  type = object({
+    app_name          = string
+    model             = string
+    config            = optional(map(string), {})
+    channel           = optional(string, "2/stable")
+    base              = optional(string, "ubuntu@22.04")
+    revision          = optional(string, null)
+    units             = optional(number, 3)
+    constraints       = optional(string, "arch=amd64")
+    machines          = optional(list(string), [])
+    storage           = optional(map(string), {})
+    endpoint_bindings = optional(map(string), {})
+  })
+}
+
+variable "failover" {
+  description = "Failover orchestrator app definition"
+  type = object({
+    app_name          = string
+    model             = string
+    config            = optional(map(string), { "init_hold" : "true" })
+    channel           = optional(string, "2/stable")
+    base              = optional(string, "ubuntu@22.04")
+    revision          = optional(string, null)
+    units             = optional(number, 3)
+    constraints       = optional(string, "arch=amd64")
+    machines          = optional(list(string), [])
+    storage           = optional(map(string), {})
+    endpoint_bindings = optional(map(string), {})
+  })
+  default = null
```

Comment:
defaults in the variables you define in your product modules.

So that way someone can "just go" without defining them themselves. It also leads to a friendly UX because they know what it should look like when they generate them themselves

Reply:
By defaulting to null, it also means that this variable is optional, and the user doesn't need to explicitly pass it.
Should they however decide to pass it, only 2 fields are mandatory "app_name / model"

Reply:
Ah TIL, so I assume it isn't possible to specify that the variable is optional AND provide an example?

Reply:
Not as far as I know, the way I know of is through "description" which is already done in the `charm/simple_deployment/` or for the non-opensearch related variables

Reply:
Thanks for the context!